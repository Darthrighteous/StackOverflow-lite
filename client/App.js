import React, { Fragment } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import store from './store';
import Header from './containers/Header';
import Spinner from './containers/Spinner';

import Home from './components/Home';
import Signup from './components/Signup';
import Login from './components/Login';
import NotFound from './components/NotFound';

const App = () => (
  <Provider store={store}>
    <Router>
      <Fragment>
        <Spinner />
        <Header />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/signup" component={Signup} />
          <Route path="/login" component={Login} />
          {/* <Route path="/new-question" component={NewQuestion} /> */}
          {/* <Route path="/question/:id" component={Question} /> */}
          <Route component={NotFound} />
        </Switch>
      </Fragment>
    </Router>
  </Provider>
);

export default App;
