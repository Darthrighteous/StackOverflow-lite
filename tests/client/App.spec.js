import React from 'react';
import { shallow } from 'enzyme';

import App from '../../client/App';

test('App snapshot test', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
