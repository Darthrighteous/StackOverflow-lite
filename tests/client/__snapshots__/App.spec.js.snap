// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Connect(Spinner) />
    <Connect(Header) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</Provider>
`;
